maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java

```diff
@@ -144,43 +148,27 @@
 
         if ( !request.getSelectedProjects().isEmpty() )
         {
-            result = new ArrayList<>( projects.size() );
+            File reactorDirectory = getReactorDirectory( request );
 
-            File reactorDirectory = null;
-            if ( request.getBaseDirectory() != null )
-            {
-                reactorDirectory = new File( request.getBaseDirectory() );
-            }
-
-            Collection<MavenProject> selectedProjects = new LinkedHashSet<>( projects.size() );
+            Collection<MavenProject> selectedProjects = new LinkedHashSet<>( request.getSelectedProjects().size() );
```

Comment:
Why change the initial size of the `selectedProjects`? It used to be `projects.size()`, now it's `request.getSelectedProjects().size()`.

Reply:
Since this set will only get as large as the amount of selected projects (-pl). I should maybe go for this amount * 2 in order for Java not unnecessarily expanding the list. What do you think?

Reply:
I've seen this slight over-allocation a lot in the Maven codebase. Personally, it feels like premature optimisation, and arguing that it'll never grow _over_ `selectedProjects.size()` makes perfect sense to me. Although you should specify the `loadFactor` to be 1.0 as well in that case, otherwise it'll still grow since the default `loadFactor` is 0.75.

Reply:
Sounds like a good solution! Since we are initializing the set with an initial capacity, I think setting the loadfactor to 1 makes perfect sense. Committed the change!